provider/provider.go

```diff
@@ -19,6 +20,8 @@ import (
 type Client interface {
 	Capabilities() ([]string, error)
 
+	TemplateContext() (openapi3.Schema, error)
```

Comment:
I wonder if this should be a part of the Capabilities return and if we just make that struct. I could also see it being helpful to have a context per capability.

Reply:
I like that idea